src/features/twitter/controllers/twitter-controller.test.ts

```diff
@@ -10,49 +10,46 @@ import { AuthorizeTwitterService } from '../services/authorize-twitter-service';
 import type { TwitterService } from '../services/twitter-service';
 import { TwitterController } from './twitter-controller';
 
-const makeSut = () => {
-  const mockLogger: Logger = mock<Logger>(loggerMock);
-  const twitterServiceMock = mock<TwitterService>({
-    getTwitterOAuthToken: vi.fn(),
-    getTwitterUser: vi.fn(),
-  });
+describe('[Controller] Twitter', () => {
+  let mockLogger: Logger;
+  let twitterServiceMock: TwitterService;
+  let authorizeTwitterService: AuthorizeTwitterService;
+  let authController: TwitterController;
+  let req: Request;
+  let res: Response;
+  let next: NextFunction;
+  beforeEach(() => {
+    mockLogger = mock<Logger>(loggerMock);
 
-  const authorizeTwitterService = mock<AuthorizeTwitterService>(
-    new AuthorizeTwitterService(
-      mockLogger,
-      twitterServiceMock,
-      accountRepositoryMock,
-      tokenRepositoryMock
-    )
-  );
+    twitterServiceMock = mock<TwitterService>({
+      getTwitterOAuthToken: vi.fn(),
+      getTwitterUser: vi.fn(),
+    });
 
-  const authController = new TwitterController(authorizeTwitterService);
+    authorizeTwitterService = mock<AuthorizeTwitterService>(
+      new AuthorizeTwitterService(
+        mockLogger,
+        twitterServiceMock,
+        accountRepositoryMock,
+        tokenRepositoryMock
+      )
+    );
 
-  const req = mockDeep<Request>();
-  const res = {
-    json: vi.fn(),
-    send: vi.fn(),
-    status: vi.fn().mockReturnThis(),
-  } as unknown as Response;
-  const next = vi.fn();
+    authController = new TwitterController(authorizeTwitterService);
 
-  return {
-    authController,
-    authorizeTwitterService,
-    mockLogger,
-    next,
-    req,
-    res,
-    twitterServiceMock,
-  };
-};
+    req = mockDeep<Request>();
+
+    res = {
+      json: vi.fn(),
+      send: vi.fn(),
+      status: vi.fn().mockReturnThis(),
+    } as unknown as Response;
+
+    next = vi.fn() as unknown as NextFunction;
+  });
```

Comment:
Uma boa prática que to aprendendo é, depois de um **beforeEach** adicionar um **afterEach** com **vi.clearAllMocks().** 

Isso garante que todos os mocks sejam resetados após cada teste, evitando efeitos colaterais entre eles. 

_"Esse padrão melhora a confiabilidade dos testes, garantindo que cada um seja executado em um estado limpo e independente dos outros, o que evita problemas difíceis de identificar, como falsos positivos ou falsos negativos devido a interferências entre os testes."_

```
afterEach(() => {
    vi.clearAllMocks();
  });
```

Se fizer sentido pra voce! <3

![happy-toddlers-and-tiaras-gif](https://github.com/user-attachments/assets/d19ff818-137e-4e14-8be2-1848e333c167)

Reply:
_O afterEach() deve ficar logo abaixo do beforeEach() e fora dos blocos describe de testes específicos, garantindo que todos os mocks sejam resetados após cada teste._

Reply:
Faz sentindo sim mano, bem observado. Como você mesmo falou da mais certeza no teste